operator/api/v1alpha1/keptntask_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KeptnTaskSpec defines the desired state of KeptnTask
+type KeptnTaskSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of KeptnTask. Edit keptntask_types.go to remove/update
+	Workload         string           `json:"workload"`
+	WorkloadVersion  string           `json:"workloadVersion"`
+	Application      string           `json:"application"`
+	TaskDefinition   string           `json:"taskDefinition"`
+	Parameters       TaskParams       `json:"parameters,omitempty"`
+	SecureParameters SecureParameters `json:"secureParameters,omitempty"`
```

Comment:
to stay more consistent with K8s naming, should we use something like `envFrom` where we can either reference a secret (`secretRef`) or configMap `configMapRef`)?

Reply:
At the moment, I'd like to keep it this way ... We should definitely discuss this in the Final Version ...